particles/Tutorial/Kotlin/Demo/TTTRandomComputer.kt

```diff
@@ -0,0 +1,48 @@
+package arcs.tutorials.tictactoe
+
+import arcs.Handle
+import arcs.Particle
+import arcs.Singleton
+import arcs.TTTRandomComputer_GameState
+import arcs.TTTRandomComputer_MyMove
+import arcs.TTTRandomComputer_Player
+import kotlin.native.internal.ExportForCppRuntime
+
+class TTTRandomComputer : Particle() {
+    private val gameState = Singleton { TTTRandomComputer_GameState() }
+    private val myMove = Singleton { TTTRandomComputer_MyMove() }
+    private val player = Singleton { TTTRandomComputer_Player() }
+
+    init {
+        registerHandle("gameState", gameState)
+        registerHandle("myMove", myMove)
+        registerHandle("player", player)
+    }
+
+    override fun onHandleSync(handle: Handle, allSynced: Boolean) {
+        onHandleUpdate(gameState)
+    }
+
+    override fun onHandleUpdate(handle: Handle) {
+        if (gameState.get()?.currentPlayer == player.get()?.id) {
```

Comment:
Consider inverting this if condition and returning (ie. turning it into a guard), so you can un-indent the actual body.